src/unicode.cpp

```diff
@@ -60,22 +60,20 @@ ada_really_inline bool has_tabs_or_newline(
   }
   // fast path for long strings (expected to be common)
   size_t i = 0;
-  const uint8x16_t mask1 = vmovq_n_u8('\r');
-  const uint8x16_t mask2 = vmovq_n_u8('\n');
-  const uint8x16_t mask3 = vmovq_n_u8('\t');
+  static uint8_t rnt_array[16] = {1, 0,  0, 0, 0,  0, 0, 0,
+                                  9, 10, 0, 0, 13, 0, 0, 0};
+  const uint8x16_t rnt = vld1q_u8(rnt_array);
+  // m['0xd', '0xa', '0x9']
   uint8x16_t running{0};
   for (; i + 15 < user_input.size(); i += 16) {
     uint8x16_t word = vld1q_u8((const uint8_t*)user_input.data() + i);
-    running = vorrq_u8(vorrq_u8(running, vorrq_u8(vceqq_u8(word, mask1),
-                                                  vceqq_u8(word, mask2))),
-                       vceqq_u8(word, mask3));
+
+    running = vorrq_u8(running, vceqq_u8(vqtbl1q_u8(rnt, word), word));
   }
   if (i < user_input.size()) {
     uint8x16_t word =
         vld1q_u8((const uint8_t*)user_input.data() + user_input.length() - 16);
-    running = vorrq_u8(vorrq_u8(running, vorrq_u8(vceqq_u8(word, mask1),
-                                                  vceqq_u8(word, mask2))),
-                       vceqq_u8(word, mask3));
+    running = vorrq_u8(running, vceqq_u8(vqtbl1q_u8(rnt, word), word));
   }
   return vmaxvq_u8(running) != 0;
```

Comment:
I don't know if that's important, but for `any` - you don't need `u8` - `max_u64` will do.

Reply:
According to @dougallj, on Apple Silicon, this would make no difference, at least as far as the latency of the SIMD instruction is concerned. I assume that the same is true on other platforms.

Reply:
Unfortunately some other cores have much less consistent reductions, for example the Cortex-A78: https://developer.arm.com/documentation/102160/latest/

"ASIMD max/min, reduce, 16B" is listed as latency 4, throughput 1/2,
"ASIMD max/min, reduce, 4H/4S" is listed as latency 2, throughput 1, and
"ASIMD max/min, basic and pairwise" is listed as latency 2, throughput 2.

Reply:
Same with Neoverse. Ok. I will change the code.